### DISCUSSION TITLE
"pop" to previous location

### DISCUSSION THREAD

**Discussion by smason** [helix-editor/helix#1981] [Q&A] [answered]
having used Emacs before Helix, I'm used to being able to return to my previous location after using, e.g., goto definition.  For example, I'd hit  `M-.` to browse to definition, then be able to hit `M-,` (i.e. `xref-pop-marker-stack`) to quickly return to where I was.

Browsing to definition is obviously easy in Helix, `gd`, but I'm wondering how to easily get back again?

**Comment by the-mikedavis** [comment, marked as answer]
For jumping to locations you'll want to use the jumplist. In ~insert~ normal mode: <kbd>Ctrl</kbd><kbd>s</kbd> saves your current position in the jump list, <kbd>Ctrl</kbd><kbd>i</kbd> allows you to go forward and <kbd>Ctrl</kbd><kbd>o</kbd> lets you jump back. Some motions will automatically save your position in the jumplist.

**Reply by smason** [reply]
ah, so that's what the "jumplist" is!  also noticed it's a term vim users will probably know

is it worth me submitting a pull request adding a note to `keymap.md` for users migrating from other editors?  I don't see any other mentions of the term elsewhere in the "book"

**Reply by the-mikedavis** [reply]
I think this would be a good fit for the wiki, probably somewhere around [here](https://github.com/helix-editor/helix/wiki/FAQ#how-to). What do you think?

**Reply by smason** [reply]
was thinking discoverability would be easier if it was nearer the place it was used, but happy to put something in the FAQ if you think that's more appropriate!

**Reply by the-mikedavis** [reply]
Hmm actually I think a small blurb below that normal mode movement section could be a good addition. There's some precedent with the [section on select mode](https://docs.helix-editor.com/keymap.html#select--extend-mode)

**Reply by leeola** [reply]
I've been wanting this for quite a while _(in any editor) and had no idea what it would be called. Really cool to see it here!

Is it possible to manage several jump lists? Ie lets say i want to jump around between errors and want that to be a separate jumplist from when i jump sourcecode via LSP?

**Reply by the-mikedavis** [reply]
For now I think the jumplist is only a single list. I think it could be cool to have one that works as a tree too although it would be harder to visualize. Those might be good plugin ideas.

**Reply by sudormrfbin** [reply]
@leeola the particular use case you mentioned would be better served by a vim-like quickfix list to which all errors can be sent to. Also IIRC we store jumps internally as a tree but only expose a linear keybind API to it ? I'm not sure though.

Edit: Nope the jumplist is stored as a simple `Vec`, nothing fancy.
